src/glum/_glm.py

```diff
@@ -1,47 +1,19 @@
-"""
-Generalized Linear Models with Exponential Dispersion Family
-
-Modified from code submitted as a PR to sklearn:
-https://github.com/scikit-learn/scikit-learn/pull/9405
-
-Original attribution from:
-https://github.com/scikit-learn/scikit-learn/pull/9405/files
-#diff-38e412190dc50455611b75cfcf2d002713dcf6d537a78b9a22cc6b1c164390d1
-'''
-Author: Christian Lorentzen <lorentzen.ch@googlemail.com>
-some parts and tricks stolen from other sklearn files.
-'''
-"""
-
-# License: BSD 3 clause
-import copy
 import re
 import sys
+import time
+import typing
 import warnings
-from collections.abc import Iterable, Mapping, Sequence
-from time import perf_counter
-from typing import Any, NamedTuple, Optional, Union, cast
+from collections.abc import Mapping, Sequence
+from typing import Any, Optional, Union
 
+import formulaic
 import numpy as np
+import packaging.version
 import pandas as pd
 import scipy.sparse as sps
-import scipy.sparse.linalg as splinalg
 import sklearn as skl
 import tabmat as tm
-from formulaic import Formula, FormulaSpec
-from formulaic.parser import DefaultFormulaParser
-from formulaic.utils.constraints import LinearConstraintParser
-from formulaic.utils.context import capture_context
-from packaging import version
 from scipy import linalg, sparse, stats
-from sklearn.base import BaseEstimator, RegressorMixin
```

Comment:
I think these imports are quite idiomatic, and that's how they appear in scikit-learn's docs, too.